Doppler.BillingUser/Controllers/WebhooksController.cs

```diff
@@ -0,0 +1,91 @@
+using Doppler.BillingUser.Enums;
+using Doppler.BillingUser.ExternalServices.MercadoPagoApi;
+using Doppler.BillingUser.Infrastructure;
+using Doppler.BillingUser.Mappers;
+using Doppler.BillingUser.Mappers.PaymentStatus;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using System.Threading.Tasks;
+
+namespace Doppler.BillingUser.Controllers
+{
+    [ApiController]
+    public class WebhooksController : ControllerBase
+    {
+        private readonly IBillingRepository _billingRepository;
+        private readonly ICurrencyRepository _currencyRepository;
+        private readonly ILogger<WebhooksController> _logger;
+        private readonly IUserRepository _userRepository;
+        private readonly IMercadoPagoService _mercadoPagoService;
+        private readonly IPaymentStatusMapper _paymentStatusMapper;
+        private readonly string PAYMENT_UPDATED = "payment.updated";
+
+        public WebhooksController(
+            ICurrencyRepository currencyRepository,
+            ILogger<WebhooksController> logger,
+            IBillingRepository billingRepository,
+            IUserRepository userRepository,
+            IMercadoPagoService mercadoPagoService,
+            IPaymentStatusMapper paymentStatusMapper)
+        {
+            _billingRepository = billingRepository;
+            _currencyRepository = currencyRepository;
+            _logger = logger;
+            _userRepository = userRepository;
+            _mercadoPagoService = mercadoPagoService;
+            _paymentStatusMapper = paymentStatusMapper;
+        }
+
+        [HttpPost("/accounts/{accountname}/integration/mercadopagonotification")]
+        public async Task<IActionResult> UpdateMercadoPagoPaymentStatusAsync([FromRoute] string accountname, [FromBody] MercadoPagoNotification notification)
+        {
+            if (notification.Action != PAYMENT_UPDATED)
+            {
+                return new OkObjectResult("Successful");
+            }
+
+            var user = await _userRepository.GetUserInformation(accountname);
+            if (user is null)
+            {
+                return new NotFoundObjectResult("Account not found");
+            }
+
+            var invoice = await _billingRepository.GetInvoice(user.IdUser, notification.Data.Id.ToString());
+            if (invoice is null)
+            {
+                _logger.LogError("Invoice with authorization number: {0} was not found.", notification.Data.Id);
```

Comment:
Rename the placeholder. See: https://docs.microsoft.com/en-US/dotnet/fundamentals/code-analysis/quality-rules/ca2253
```suggestion
                _logger.LogError("Invoice with authorization number: {authorizationNumber} was not found.", notification.Data.Id);
```